File: CS50x - Intro to CompSci/credit/credit.c
// Include libraries
#include <cs50.h>
#include <math.h>
#include <stdio.h>

// Function to remove digits
long removeDigits(long number, long digits)
{
    return ((number - (number % (long) pow(10, digits))) / (long) pow(10, digits));
}

int main(void)
{
    // Get credit card number
    long number = get_long("Number: ");

    // Check if starting is invalid and if the number is too long or short
    if (removeDigits(number, 13) != 34 &&
        removeDigits(number, 13) != 37 &&
        removeDigits(number, 14) != 51 &&
        removeDigits(number, 14) != 52 &&
        removeDigits(number, 14) != 53 &&
        removeDigits(number, 14) != 54 &&
        removeDigits(number, 14) != 55 &&
        removeDigits(number, 12) != 4 &&
        removeDigits(number, 15) != 4)
    {
        printf("INVALID\n");
        return 0;
    }

    // Perform Luhn's algorithm
    long sumOfProductsDigits = 0;
    for (long a = 0; a < 9; a++)
    {
        for (long b = 0; b < 2; b++)
        {
            sumOfProductsDigits += removeDigits((removeDigits(number, 2 * a + 1) % 10) * 2, b) % 10;
        }
    }
    for (long c = 0; c < 9; c++)
    {
        sumOfProductsDigits += removeDigits(number, 2 * c) % 10;
    }
    if (sumOfProductsDigits % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }

    // Find type of card
    if (removeDigits(number, 13) == 34 ||
        removeDigits(number, 13) == 37)
    {
        printf("AMEX\n");
        return 0;
    }
    if (removeDigits(number, 14) == 51 ||
        removeDigits(number, 14) == 52 ||
        removeDigits(number, 14) == 53 ||
        removeDigits(number, 14) == 54 ||
        removeDigits(number, 14) == 55)
    {
        printf("MASTERCARD\n");
        return 0;
    }
    if (removeDigits(number, 12) == 4 ||
        removeDigits(number, 15) == 4)
    {
        printf("VISA\n");
        return 0;
    }
}

File: CS50x - Intro to CompSci/sentimental-credit/credit.py
def removeDigits(number, digits):
    return int((number - (number % (10**digits))) / (10**digits))


def numberI():
    n = input("Number: ")
    try:
        n = int(n)
    except:
        return numberI()
    else:
        return n


number = numberI()
if (removeDigits(number, 13) != 34 and removeDigits(number, 13) != 37 and removeDigits(number, 14) != 51 and removeDigits(number, 14) != 52 and removeDigits(number, 14) != 53 and removeDigits(number, 14) != 54 and removeDigits(number, 14) != 55 and removeDigits(number, 12) != 4 and removeDigits(number, 15) != 4):
    print("INVALID")
    quit()
sumOfProductsDigits = 0
for a in range(9):
    for b in range(2):
        sumOfProductsDigits += removeDigits((removeDigits(number, 2 * a + 1) % 10) * 2, b) % 10
for c in range(9):
    sumOfProductsDigits += removeDigits(number, 2 * c) % 10
if sumOfProductsDigits % 10 != 0:
    print("INVALID")
    quit()
if (removeDigits(number, 13) == 34 or removeDigits(number, 13) == 37):
    print("AMEX")
    quit()
if (removeDigits(number, 14) == 51 or removeDigits(number, 14) == 52 or removeDigits(number, 14) == 53 or removeDigits(number, 14) == 54 or removeDigits(number, 14) == 55):
    print("MASTERCARD")
    quit()
if (removeDigits(number, 12) == 4 or removeDigits(number, 15) == 4):
    print("VISA")
    quit()